crates/ruff_linter/src/rules/ruff/rules/ambiguous_unicode_character.rs

```diff
@@ -171,16 +173,59 @@ impl Violation for AmbiguousUnicodeCharacterComment {
     }
 }
 
-/// RUF001, RUF002, RUF003
-pub(crate) fn ambiguous_unicode_character(
+/// RUF003
+pub(crate) fn ambiguous_unicode_character_comment(
     diagnostics: &mut Vec<Diagnostic>,
     locator: &Locator,
     range: TextRange,
-    context: Context,
     settings: &LinterSettings,
 ) {
     let text = locator.slice(range);
+    ambiguous_unicode_character(diagnostics, text, range, Context::Comment, settings);
+}
 
+/// RUF001, RUF002
+pub(crate) fn ambiguous_unicode_character_string(checker: &mut Checker, string_like: StringLike) {
+    let context = if checker.semantic().in_docstring() {
+        Context::Docstring
+    } else {
+        Context::String
+    };
+
+    match string_like {
+        StringLike::StringLiteral(string_literal) => {
+            for string in &string_literal.value {
+                let text = checker.locator().slice(string);
+                ambiguous_unicode_character(
```

Comment:
I think we could reduce the arguments necessary to pass to the function when the function returns the Vec and we move the generation of the diagnostics into this function

Reply:
Do you mean that we allocate a vector of diagnostics inside the function and then pass it outside to be extended onto the global diagnostics vector?

Reply:
The idea was to return the `word_candidates` vec if `word_flags.is_candidate_word()` and otherwise return an empty `Vec`. You can then generate the `Diagnostics` in the call site. The downside is that it will result in some duplication. I wasn't aware that we call this methods for comments too.

Reply:
I see. I'll continue as is for now because of how `StringLike` enum is. We can revisit later.